python/servo/devenv_commands.py

```diff
@@ -284,38 +274,32 @@ def ndk_gdb(self, release, target):
 
     @Command('try',
-             description='Runs try jobs by force pushing to personal fork try branches',
+             description='Runs try jobs by force pushing to try branch',
              category='devenv')
     @CommandArgument(
-        'jobs', default=["try"], nargs='...',
-        help="Name(s) of job(s) (ex: try, linux, mac, windows, wpt)")
-    def try_jobs(self, jobs):
-        branches = []
-        # we validate branches because force pushing is destructive
-        for job in jobs:
-            # branches must start with try-
-            if "try" not in job:
-                job = "try-" + job
-            if job not in VALID_TRY_BRACHES:
-                print(job + " job doesn't exist")
-                return -1
-            branches.append(job)
-        remote = "origin"
-        if "servo/servo" in subprocess.check_output(["git", "config", "--get", "remote.origin.url"]).decode():
-            # if we have servo/servo for origin check try remote
-            try:
-                if "servo/servo" in subprocess.check_output(["git", "config", "--get", "remote.try.url"]).decode():
-                    # User has servo/servo for try remote
-                    print("You should not use servo/servo for try remote!")
-                    return -1
-                else:
-                    remote = "try"
-            except subprocess.CalledProcessError:
-                print("It looks like you are patching in upstream servo.")
-                print("Set try remote to your personal fork with `git remote add try https://github.com/user/servo`")
-                return -1
-        for b in branches:
-            res = call(["git", "push", remote, "--force", f"HEAD:{b}"], env=self.build_env())
+        '--remote', '-r', default="origin",
+        help='git remote to use for try pushes')
+    @CommandArgument(
+        'try_string', default=None, nargs='...',
+        help="Try string")
+    def try_jobs(self, remote="origin", try_string=None):
+        # if jobs is empty we just push last commit to try and it will do full run
+        if try_string:
+            try_string = " ".join(try_string)
+            sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "..", "etc"))
+            from python.try_parser import Config
+            conf = Config(try_string)
+            res = call(["git", "commit", "--allow-empty", "-m", f"{try_string}\n\n{conf.toJSON()}"],
+                       env=self.build_env())
             if res != 0:
                 return res
-        return 0
+
+        if "servo/servo" in subprocess.check_output(["git", "config", "--get", f"remote.{remote}.url"]).decode():
+            print("WARNING: You are triggering try build in upstream repo!")
+
+        res = call(["git", "push", remote, "--force", "HEAD:try"], env=self.build_env())
+
+        if try_string:
+            # we need to delete last commit
+            res += call(["git", "reset", "--soft", "HEAD~1"], env=self.build_env())
```

Comment:
I'm not sure this is necessary. I think it's fine to leave the last commit in the try branch.

Reply:
This is necessary as we add commit on top of current checkout (HEAD) then we push this to remote try branch, so now we need to remove try commit from current checkout, like it was never there.

Reply:
Oh, I get it now. Thanks for the clarification.